src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueTests.cs

```diff
@@ -3511,77 +3539,87 @@ class C
+            var baseParameterCount = reader0.GetParameters().Count();
+
+            Assert.Equal(3, baseTypeCount);
+            Assert.Equal(4, baseMethodCount);
+            Assert.Equal(7, baseAttributeCount);
+            Assert.Equal(2, baseParameterCount);
+
             // This update emulates "Reloadable" type behavior - a new type is generated instead of updating the existing one.
             var diff1 = compilation1.EmitDifference(
                 generation0,
                 ImmutableArray.Create(
                     SemanticEdit.Create(SemanticEditKind.Replace, null, c1)));
 
-            // Verify delta metadata contains expected rows.
             using var md1 = diff1.GetMetadata();
-            var reader1 = md1.Reader;
-            var readers = new[] { reader0, reader1 };
-
-            CheckNames(readers, reader1.GetTypeDefNames(), "C#1");
-            CheckNames(readers, diff1.EmitResult.ChangedTypes, "C#1");
-
-            CheckEncLogDefinitions(reader1,
-                Row(3, TableIndex.TypeDef, EditAndContinueOperation.Default),
-                Row(3, TableIndex.TypeDef, EditAndContinueOperation.AddMethod),
-                Row(3, TableIndex.MethodDef, EditAndContinueOperation.Default),
-                Row(3, TableIndex.TypeDef, EditAndContinueOperation.AddMethod),
-                Row(4, TableIndex.MethodDef, EditAndContinueOperation.Default),
-                Row(3, TableIndex.MethodDef, EditAndContinueOperation.AddParameter),
-                Row(2, TableIndex.Param, EditAndContinueOperation.Default),
-                Row(3, TableIndex.MethodDef, EditAndContinueOperation.AddParameter),
-                Row(3, TableIndex.Param, EditAndContinueOperation.Default));
+            ValidateReplacedType(diff1, new[] { reader0, md1.Reader });
 
-            CheckEncMapDefinitions(reader1,
-                Handle(3, TableIndex.TypeDef),
-                Handle(3, TableIndex.MethodDef),
-                Handle(4, TableIndex.MethodDef),
-                Handle(2, TableIndex.Param),
-                Handle(3, TableIndex.Param));
-
-            // This update emulates "Reloadable" type behavior - a new type is generated instead of updating the existing one.
             var diff2 = compilation2.EmitDifference(
                 diff1.NextGeneration,
                 ImmutableArray.Create(
                     SemanticEdit.Create(SemanticEditKind.Replace, null, c2)));
 
-            // Verify delta metadata contains expected rows.
             using var md2 = diff2.GetMetadata();
-            var reader2 = md2.Reader;
-            readers = new[] { reader0, reader1, reader2 };
+            ValidateReplacedType(diff2, new[] { reader0, md1.Reader, md2.Reader });
 
-            CheckNames(readers, reader2.GetTypeDefNames(), "C#2");
-            CheckNames(readers, diff2.EmitResult.ChangedTypes, "C#2");
+            void ValidateReplacedType(CompilationDifference diff, MetadataReader[] readers)
```

Comment:
nit: when possible we try to put local functions at the end of the method so as to keep the main logic uninterrupted

Reply:
In this case I believe this layout is better since the validation code only applies to and is very specific to generation \#1 and \#2 and not the last one. I'm actually gonna move the code a few lines higher, so that I can write matching code in VB using lambda.